sdk/contentsafety/azure-ai-contentsafety/README.md

```diff
@@ -26,26 +26,54 @@ You can find the endpoint for your Azure AI Content Safety service resource usin
 az cognitiveservices account show --name "resource-name" --resource-group "resource-group-name" --query "properties.endpoint"
 ```
 
-#### Get the API key
+#### Create a ContentSafetyClient with API key
 
+To use an API key as the `credential` parameter.
+
+- Step 1: Get the API key. 
 The API key can be found in the [Azure Portal][azure_portal] or by running the following [Azure CLI][azure_cli_key_lookup] command:
 
-```bash
-az cognitiveservices account keys list --name "<resource-name>" --resource-group "<resource-group-name>"
-```
+    ```bash
+    az cognitiveservices account keys list --name "<resource-name>" --resource-group "<resource-group-name>"
+    ```
+
+- Step 2: Pass the key as a string into an instance of `AzureKeyCredential`.
 
-#### Create a ContentSafetyClient with AzureKeyCredential
+    ```python
+    from azure.core.credentials import AzureKeyCredential
+    from azure.ai.contentsafety import ContentSafetyClient
+    
+    endpoint = "https://<my-custom-subdomain>.cognitiveservices.azure.com/"
+    credential = AzureKeyCredential("<api_key>")
+    client = ContentSafetyClient(endpoint, credential)
+    ```
 
-To use an API key as the `credential` parameter, pass the key as a string into an instance of `AzureKeyCredential`.
+#### Create a ContentSafetyClient with Service Principal
 
-```python
-from azure.core.credentials import AzureKeyCredential
-from azure.ai.contentsafety import ContentSafetyClient
+To use a service principal as the `credential` parameter:
 
-endpoint = "https://<my-custom-subdomain>.cognitiveservices.azure.com/"
-credential = AzureKeyCredential("<api_key>")
-client = ContentSafetyClient(endpoint, credential)
-```
+ - Step 1: Create and register a Microsoft Entra application, see [Create a Microsoft Entra app registration.](https://learn.microsoft.com/en-us/azure/data-explorer/provision-entra-id-app#create-microsoft-entra-application-registration) 
+And you may need to record the `tenant_id`, `client_id` and `client_secret` of this application.
+
+ - Step 2: Assign roles.
+   - In the Azure portal, find your Azure AI Content Safety resource.
+   - Select **Access Control(IAM)**
+   - Select **+ Add**, and then select **Add role assignment** from the menu.
+   - Set the role to **Cognitive Service User** and select the service principal you just created as member. Then select Save.
```

Comment:
It's not suggested to write detailed steps here since content in the link may change. You can simplify it like this:

Step 1: Enable AAD for your resource
Please refer to this Cognitive Services authentication document(https://learn.microsoft.com/en-us/azure/ai-services/authentication?tabs=powershell#authenticate-with-microsoft-entra-id) for the steps to enable AAD for your resource. The main steps are:

- Create resource with a custom subdomain.
- Create Service Principal and assign Cognitive Services User role to it.